admin/category_product_listing.php

```diff
@@ -64,44 +64,46 @@
 
         // change the status of categories and products
         zen_set_time_limit(600);
-        for ($i = 0, $n = sizeof($categories); $i < $n; $i++) {
           if ($_POST['categories_status'] == '1') {//form is coming from an Enabled category which is to be changed to Disabled
-            $categories_status = '0';//Disable this category
-            $update_subcategories = $_POST['set_subcategories_status'] == 'set_subcategories_status_off' ? '1' : '0'; //also Disable subcategories?
+              $category_status = '0';//Disable this category
+              $subcategories_status = isset($_POST['set_subcategories_status']) && $_POST['set_subcategories_status'] == 'set_subcategories_status_off' ? '0' : ''; //Disable subcategories or no change?
+              $products_status = isset($_POST['set_products_status']) && $_POST['set_products_status'] == 'set_products_status_off' ? '0' : ''; //Disable products or no change?
           } else {//form is coming from a Disabled category which is to be changed to Enabled
-            $categories_status = '1';//Enable this category
-            $update_subcategories = $_POST['set_subcategories_status'] == 'set_subcategories_status_on' ? '1' : '0'; //also Enable subcategories?
+              $category_status = '1';//Enable this category
+              $subcategories_status = isset($_POST['set_subcategories_status']) && $_POST['set_subcategories_status'] == 'set_subcategories_status_on' ? '1' : ''; //also Enable subcategories or no change?
+              $products_status = isset($_POST['set_products_status']) && $_POST['set_products_status'] == 'set_products_status_on' ? '1' : ''; //Disable products or no change?
           }
 
-          if ($categories[$i]['id'] == $categories_id || $update_subcategories) {//always update THIS category, optionally update subcategories
-            $sql = "UPDATE " . TABLE_CATEGORIES . "
-                    SET categories_status = " . (int)$categories_status . "
+        for ($i = 0, $n = sizeof($categories); $i < $n; $i++) {
+
+          //set categories_status
+            if ($categories[$i]['id'] == $categories_id) {//always update THIS category
+                $sql = "UPDATE " . TABLE_CATEGORIES . "
+                    SET categories_status = " . $category_status . "
+                    WHERE categories_id = " . $categories[$i]['id'];
+                $db->Execute($sql);
```

Comment:
Why have you removed the typecasting here?

Reply:
Since it was set to '1' a few lines above I thought it didn't need it. I'll put it back.